packages/design-system/src/components/Table/TableCaption.tsx

```diff
@@ -1,15 +1,37 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import classNames from 'classnames';
 
-export const TableCaption = ({
+export interface TableCaptionProps {
+  /**
+   * The table caption contents.
+   */
+  children?: React.ReactNode;
+  /**
+   * Additional classes to be added to the caption element.
+   */
+  className?: string;
+  /**
+   * @hide-prop This gets passed from the parent `Table` component when the table `scrollable` prop is set.
+   */
+  _id?: string;
+  /**
+   * @hide-prop This gets passed from the parent `Table` component when the table `scrollable` prop is set.
+   */
+  _scrollActive?: boolean;
+  /**
+   * @hide-prop This gets passed from the parent `Table` component when the table `scrollable` prop is set.
+   */
+  _scrollableNotice?: React.ReactNode;
+}
+
+export const TableCaption: React.FC<TableCaptionProps> = ({
```

Comment:
[question] in what situation does the component need to omit the `children` prop (like in TableRow component) vs this component where it is not omitted?

Reply:
In the case of `TableCaption` we don't use `React.ComponentPropsWithRef` and just refer to TableCaptionProps, so there's no need to omit any attributes.

For components which has `React.ComponentPropsWithRef` or `React.ComponentPropsWithoutRef` of an html element, and if the html element has the same attribute as what's on our component's prop (eg. TableRowProps has children), then we want to omit these attribute and use ours instead. 

Also if we don't omit `children` in the case of TableRow, the children prop (see below image) will not show up on the props table of our document site.
![Screen Shot 2021-07-20 at 1 11 35 PM](https://user-images.githubusercontent.com/20322880/126366808-a2889366-a40d-4052-93ce-6a287947026e.png)